cloudsync/tests/test_cs.py

```diff
@@ -819,6 +819,35 @@ def test_cs_basic(cs):
     assert not cs.state.changeset_len
 
 
+def test_cs_stop(four_local_cs):
+    syncs = list(four_local_cs)
+    for cs in syncs:
+        cs.start()
+
+    # stop a single cloudsync
+    syncs[0].stop()
+    assert syncs[0].sthread is None
+
+    # stop(join=False) + join()
+    with patch.object(syncs[1], "sthread") as sthread:
+        syncs[1]._stop(join=False)
+        assert syncs[1].sthread
+        sthread.join.assert_not_called()
+
+        syncs[1]._join()
+        assert not syncs[1].sthread
+        sthread.join.assert_called_once()
+
+    # make sure the Thread object patched above is still a Thread, so that it can be joined below
+    assert isinstance(syncs[1].sthread, threading.Thread)
+
+    # stop_all()
+    CloudSync.stop_all(syncs)
+    assert not syncs[1].sthread
```

Comment:
This probably isn't the right MR to fix this issue, but I think that setting the `sthread` to `None` is weird and leads to race conditions. I would like to get rid of that behavior entirely (the proposed new behavior being to simply leave `sthread` alone, and check `sthread.is_alive()` to determine if the thread is still going). Without actually implementing any of the new behavior, can you think of a test that doesn't further enshrine this risky behavior? Maybe grab a local copy of the sthread's and assert `not is_alive()` on them? Something like that

Reply:
sure, makes sense. don't really like using sthread as a flag either.

Reply:
got rid of sthread / ethreads